src/libraries/System.Text.Json/tests/JsonNodeOptionsTests.cs

```diff
@@ -0,0 +1,120 @@
+            Assert.ThrowsAny<JsonException>(() => JsonNode.Parse(jsonStringWithComments, options));
+
+            options = new JsonNodeOptions { CommentHandling = JsonCommentHandling.Skip };
+            JsonObject jsonObject = (JsonObject)JsonNode.Parse(jsonStringWithComments, options);
+
+            Assert.Equal("first value", jsonObject["firstProperty"]);
+            Assert.Equal("second value", jsonObject["secondProperty"]);
+
+            Assert.Equal(2, jsonObject.GetPropertyNames().Count);
+            Assert.Equal(2, jsonObject.GetPropertyValues().Count);
+        }
+
+        [Fact]
+        public static void TestTrailingCommas()
+        {
+            string jsonStringWithTrailingCommas = @"
+            {
+                ""firstProperty"": ""first value"",
+                ""secondProperty"" : ""second value"",
+            }";
+
+            JsonNodeOptions options = new JsonNodeOptions { AllowTrailingCommas = default };
+            Assert.ThrowsAny<JsonException>(() => JsonNode.Parse(jsonStringWithTrailingCommas, options));
+
+            options = new JsonNodeOptions { AllowTrailingCommas = false };
+            Assert.ThrowsAny<JsonException>(() => JsonNode.Parse(jsonStringWithTrailingCommas, options));
+
+            options = new JsonNodeOptions { AllowTrailingCommas = true };
+            JsonObject jsonObject = (JsonObject) JsonNode.Parse(jsonStringWithTrailingCommas, options);
+
+            Assert.Equal("first value", jsonObject["firstProperty"]);
+            Assert.Equal("second value", jsonObject["secondProperty"]);
+
+            Assert.Equal(2, jsonObject.GetPropertyNames().Count);
+            Assert.Equal(2, jsonObject.GetPropertyValues().Count);
+        }
+
+        [Fact]
+        public static void TestInvalidJsonNodeOptions()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => new JsonNodeOptions()
+            {
+                CommentHandling = (JsonCommentHandling)Enum.GetNames(typeof(JsonCommentHandling)).Length
+            });
+            Assert.Throws<ArgumentOutOfRangeException>(() => new JsonNodeOptions()
+            {
+                MaxDepth = -1
+            });
+            Assert.Throws<ArgumentOutOfRangeException>(() => new JsonNodeOptions()
+            {
+                DuplicatePropertyNameHandling = (DuplicatePropertyNameHandlingStrategy)Enum.GetNames(typeof(DuplicatePropertyNameHandlingStrategy)).Length
+            });
+            Assert.Throws<ArgumentOutOfRangeException>(() => new JsonNodeOptions()
+            {
+                DuplicatePropertyNameHandling = (DuplicatePropertyNameHandlingStrategy)(-1)
+            });
+        }
+
+        [Fact]
+        public static void TestDefaultJsonNodeOptions()
+        {
+            JsonNodeOptions defaultOptions = default;
+            JsonNodeOptions newOptions = new JsonNodeOptions();
+
+            Assert.Equal(defaultOptions.AllowTrailingCommas, newOptions.AllowTrailingCommas);
```

Comment:
nit: Can you also add asserts for what the `defaultOptions` are?

```C#
Assert.Equals(false, defaultOptions.AllowTrailingCommas);
Assert.Equals(JsonCommentHandling.Disallow, defaultOptions.CommentHandling);
Assert.Equals(DuplicatePropertyNameHandlingStrategy.Replace, defaultOptions.DuplicatePropertyNameHandling);
Assert.Equals(0, defaultOptions.MaxDepth);
```

Reply:
Done!